HLTrigger/Timer/plugins/ThroughputService.cc

```diff
@@ -15,41 +18,60 @@
 // describe the module's configuration
 void ThroughputService::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
   edm::ParameterSetDescription desc;
-  desc.addUntracked<double>("timeRange", 60000.0);
-  desc.addUntracked<double>("timeResolution", 10.0);
-  desc.addUntracked<std::string>("dqmPath", "HLT/Throughput");
-  desc.addUntracked<bool>("dqmPathByProcesses", false);
+  desc.addUntracked<uint32_t>("eventRange", 10000)->setComment("Preallocate a buffer for N events");
+  desc.addUntracked<uint32_t>("eventResolution", 1)->setComment("Sample the processing time every N events");
+  desc.addUntracked<bool>("printEventSummary", false);
+  desc.ifValue(edm::ParameterDescription<bool>("enableDQM", true, false),  // "false" means untracked
+               // parameters if "enableDQM" is "true"
+               true >> (edm::ParameterDescription<bool>("dqmPathByProcesses", false, false) and
+                        edm::ParameterDescription<std::string>("dqmPath", "HLT/Throughput", false) and
+                        edm::ParameterDescription<double>("timeRange", 60000.0, false) and
+                        edm::ParameterDescription<double>("timeResolution", 10.0, false)) or
+                   // parameters if "enableDQM" is "false"
+                   false >> edm::EmptyGroupDescription());
```

Comment:
I would say that the indentation here is wrong - but this is what clang-format requires